monkey/monkey_island/cc/environment/test_environment_config.py

```diff
@@ -0,0 +1,139 @@
+        "server_config": "password",
+        "deployment": "develop",
+        "user": "test",
+        "aws": "test",
+        "test": "test",
+        "test2": "test2"
+    }
+
+    CONFIG_STANDARD_ENV = {
+        "server_config": "standard",
+        "deployment": "develop"
+    }
+
+    CONFIG_STANDARD_WITH_CREDENTIALS = {
+        "server_config": "standard",
+        "deployment": "develop",
+        "user": "test",
+        "password_hash": "9ece086e9bac491fac5c1d1046ca11d737b92a2b2ebd93f005d7b710110c0a678288166e7fbe796883a"
+                         "4f2e9b3ca9f484f521d0ce464345cc1aec96779149c14"
+    }
+
+    def test_get_from_json(self):
+        self._test_get_from_json(TestEnvironmentConfig.CONFIG_WITH_CREDENTIALS)
+        self._test_get_from_json(TestEnvironmentConfig.CONFIG_NO_CREDENTIALS)
+        self._test_get_from_json(TestEnvironmentConfig.CONFIG_PARTIAL_CREDENTIALS)
+
+    def _test_get_from_json(self, config: Dict):
+        config_json = json.dumps(config)
+        env_config_object = EnvironmentConfig.get_from_json(config_json)
+        self.assertEqual(config['server_config'], env_config_object.server_config)
+        self.assertEqual(config['deployment'], env_config_object.deployment)
+        if 'user' in config:
+            self.assertEqual(config['user'], env_config_object.user_creds.username)
+        if 'password_hash' in config:
+            self.assertEqual(config['password_hash'], env_config_object.user_creds.password_hash)
+        if 'aws' in config:
+            self.assertEqual(config['aws'], env_config_object.aws)
+
+    def test_save_to_file(self):
+        self._test_save_to_file(TestEnvironmentConfig.CONFIG_WITH_CREDENTIALS)
+        self._test_save_to_file(TestEnvironmentConfig.CONFIG_NO_CREDENTIALS)
+        self._test_save_to_file(TestEnvironmentConfig.CONFIG_PARTIAL_CREDENTIALS)
+
+    @patch.object(target=EnvironmentConfig, attribute="get_config_file_path",
+                  new=MagicMock(return_value=get_server_config_file_path_test_version()))
+    def _test_save_to_file(self, config: Dict):
+        user_creds = UserCreds.get_from_dict(config)
+        env_config = EnvironmentConfig(server_config=config['server_config'],
+                                       deployment=config['deployment'],
+                                       user_creds=user_creds)
+
+        env_config.save_to_file()
+        file_path = get_server_config_file_path_test_version()
+        with open(file_path, 'r') as f:
+            content_from_file = f.read()
+        os.remove(file_path)
+
+        self.assertDictEqual(config, json.loads(content_from_file))
+
+    def test_get_server_config_file_path(self):
+        if platform.system() == "Windows":
+            server_file_path = MONKEY_ISLAND_ABS_PATH + "\cc/server_config.json"
+        else:
+            server_file_path = MONKEY_ISLAND_ABS_PATH + "/cc/server_config.json"
+        self.assertEqual(EnvironmentConfig.get_config_file_path(), server_file_path)
```

Comment:
Path equality should be done with `pathlib` and not strings

Reply:
I disagree: if test code is exactly the same as production then we'll get exactly the same results. This tests if we use pathlib correctly and if it's interface didn't change.